### PR TITLE
Add pull-request implementation for bitbucket forge 

### PR DESCRIPTION
**Pull request by mzampetakis**

### CODE REVIEW

**Comment by mzampetakis** [comment]
For the testing I run
```
GET http://DOMAIN/api/repos/10
---

{
    "id": 10,
    "forge_remote_id": "{UUID}",
    "owner": "michaliszampetakis",
    "name": "test-repo",
    "full_name": "michaliszampetakis/test-repo",
    "avatar_url": "",
    "link_url": "https://bitbucket.org/michaliszampetakis/test-repo",
    "clone_url": "https://bitbucket.org/michaliszampetakis/test-repo.git",
    "default_branch": "master",
    "scm": "git",
    "timeout": 60,
    "visibility": "private",
    "private": true,
    "trusted": false,
    "gated": false,
    "active": true,
    "allow_pr": true,
    "config_file": "",
    "cancel_previous_pipeline_events": [
        "",
        "",
        "push",
        "pull_request"
    ],
    "netrc_only_trusted": true
}
```

And then having 2 PRs open:
```
GET http://DOMAIN/api/repos/10/pull_requests
[
    {
        "index": 2,
        "title": "README.md edited online with Bitbucket"
    },
    {
        "index": 1,
        "title": "Update README.md"
    }
]
```

File: server/forge/bitbucket/bitbucket.go [hunk 1]
```diff
@@ -302,8 +302,21 @@ func (c *config) BranchHead(_ context.Context, _ *model.User, _ *model.Repo, _ s
...
-	return nil, forge_types.ErrNotImplemented
+// PullRequests returns the pull requests of the named repository.
+func (c *config) PullRequests(ctx context.Context, u *model.User, r *model.Repo, p *model.ListOptions) ([]*model.PullRequest, error) {
+	opts := internal.ListOpts{Page: p.Page, PageLen: p.Page}
+	pullRequests, err := c.newClient(ctx, u).ListPullRequests(r.Owner, r.Name, &opts)
+	if err != nil {
+		return nil, err
+	}
+	result := make([]*model.PullRequest, len(pullRequests))
+	for i := range pullRequests {
```
Comment (anbraten) [inline server/forge/bitbucket/bitbucket.go:313]: I would prefer to use this style:

```suggestion
	for _, pullRequest := range pullRequests {
```

**Review by anbraten — APPROVED** [review]

File: server/forge/bitbucket/bitbucket.go (hunk 1 above)
Comment (mzampetakis) [reply server/forge/bitbucket/bitbucket.go:313]: OK nice. Is this something generally preferred in woodpecker-ci project?
Should I update this even it's now approved?

Comment (anbraten) [reply server/forge/bitbucket/bitbucket.go:313]: I prefer (and I think most other maintainers) prefer that way. Yes, please update. Approving can be done as open threads are blocking and need to be resolved first before proceeding.

Comment (mzampetakis) [reply server/forge/bitbucket/bitbucket.go]: Done. Updated the iteration type.

**Pull request merged**